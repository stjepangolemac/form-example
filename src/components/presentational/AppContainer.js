import styled from "styled-components";

const AppContainer = styled.div`
  height: 100vh;
  display: flex;
  justify-content: center;
  padding: 20vh 0;
`;

export default AppContainer
