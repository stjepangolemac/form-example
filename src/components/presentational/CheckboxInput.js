import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Error = styled.p`
  color: red;
  font-size: 0.75em;
`;

const CheckboxInput = ({
  label,
  input,
  meta: { pristine, error, submitError }
}) => (
  <React.Fragment>
    <input {...input} type="checkbox" />
    <label htmlFor={input.id}>{label}</label>
    <Error>{!pristine && error || submitError}</Error>
  </React.Fragment>
);

CheckboxInput.propTypes = {
  input: PropTypes.shape({
    value: PropTypes.string,
    onChange: PropTypes.func
  }),
  meta: PropTypes.shape({
    error: PropTypes.string,
    submitError: PropTypes.string,
    pristine: PropTypes.bool
  }),
  label: PropTypes.string
};

CheckboxInput.defaultProps = {
  input: {},
  meta: {},
}

export default CheckboxInput;
