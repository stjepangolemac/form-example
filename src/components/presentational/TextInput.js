import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const InputContainer = styled.div`
  margin-bottom: 15px;
`;

const Input = styled.input`
  padding: 10px 5px;
  border: 1px solid lightgray;
  border-radius: 2px;
  width: 100%;
`;

const Error = styled.p`
  color: red;
  font-size: 0.75em;
`;

const Red = styled.span`
  color: red;
`;

const TextInput = ({
  label,
  type,
  input,
  meta: { pristine, error, submitError }
}) => (
  <InputContainer>
    <label htmlFor={input.id}>
      {label}: <Red>*</Red>
    </label>
    <br />
    <Input {...input} type={type || "text"} />
    <Error>{(!pristine && error) || submitError}</Error>
  </InputContainer>
);

TextInput.displayName = "TextInput";

TextInput.propTypes = {
  input: PropTypes.shape({
    value: PropTypes.string,
    onChange: PropTypes.func
  }),
  meta: PropTypes.shape({
    error: PropTypes.string,
    submitError: PropTypes.string,
    pristine: PropTypes.bool
  }),
  label: PropTypes.string,
  type: PropTypes.string,
  required: PropTypes.bool
};

TextInput.defaultProps = {
  input: {},
  meta: {}
};

export default TextInput;
