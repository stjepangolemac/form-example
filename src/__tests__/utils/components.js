import React from "react";
import renderer from "react-test-renderer";

import AppContainer from "pres/AppContainer";
import CheckboxInput from "pres/CheckboxInput";
import TextInput from "pres/TextInput";

const presComponents = [[AppContainer, {}], [CheckboxInput, {}], [TextInput, {}]];

presComponents.forEach(args => {
  test( "presentational componenets should match snapshots",
    () => {
      const Component = args[0]
      const props = args[1]

      const tree = renderer.create(<Component {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    }
  );
});
