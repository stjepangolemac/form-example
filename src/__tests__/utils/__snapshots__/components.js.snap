// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should match snapshot 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
/>
`;

exports[`Form should match snapshot 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
/>
`;

exports[`FormPage should match snapshot 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div>
    <ul
      className="FormPage__PagesNavigation-fkwltY bfNUgn"
    >
      <li
        className="FormPage__NavItem-kKpNlV chQxTR"
      >
        foo
      </li>
    </ul>
    <div
      className="FormPage__PageContent-iNekjJ jFejDm"
    >
      bar
    </div>
    <div
      className="Center-jVIFgs gUWusV"
    >
      <button
        className="Button-hYXUXp ksoPdZ"
        onClick={[Function]}
      >
        Previous
      </button>
      <button
        className="Button-hYXUXp ksoPdZ"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</form>
`;

exports[`components AppContainer component should match snapshot 1`] = `
<div
  className="AppContainer-fyEKNQ dBeOiF"
/>
`;

exports[`components Button component should match snapshot 1`] = `
<button
  className="Button-hYXUXp ceqDRH"
>
  Click me
</button>
`;

exports[`components Center component should match snapshot 1`] = `
<div
  className="Center-jVIFgs gUWusV"
>
  <div>
    foo
  </div>
</div>
`;

exports[`components CheckboxInput component should match snapshot 1`] = `
Array [
  <input
    type="checkbox"
  />,
  <label
    htmlFor={undefined}
  />,
  <p
    className="CheckboxInput__Error-ddbWLI wrdjp"
  />,
]
`;

exports[`components TextInput component should match snapshot 1`] = `
<div
  className="TextInput__InputContainer-eyrJvj fPIlSe"
>
  <label
    htmlFor={undefined}
  >
    : 
    <span
      className="TextInput__Red-eBiRuV eTAOJM"
    >
      *
    </span>
  </label>
  <br />
  <input
    className="TextInput__Input-dYZOmi gmYQdN"
    type="text"
  />
  <p
    className="TextInput__Error-bPOzPg jtXBAh"
  />
</div>
`;
